kong/plugins/aws-lambda/v4.lua

```diff
@@ -0,0 +1,227 @@
+-- Performs AWSv4 Signing
+-- http://docs.aws.amazon.com/general/latest/gr/sigv4_signing.html
+
+local resty_sha256 = require "resty.sha256"
+local pl_string = require "pl.stringx"
+local crypto = require "crypto"
+
+local ALGORITHM = "AWS4-HMAC-SHA256"
+
+local function hmac(key, msg)
+  return crypto.hmac.digest("sha256", msg, key, true)
+end
+
+local function hash(str)
+  local sha256 = resty_sha256:new()
+  sha256:update(str)
+  return sha256:final()
+end
+
+local function hex_encode(str) -- From prosody's util.hex
+  local char_to_hex = {};
```

Comment:
This `char_to_hex` table is a constant, filled right below. This one must move outside the scope (top of module?) or it will regenerate the table on every invocation